src/Maker/MakeTwigComponent.php

```diff
@@ -47,9 +51,37 @@ public function configureCommand(Command $command, InputConfiguration $inputConf
         ;
     }
 
-    public function configureDependencies(DependencyBuilder $dependencies): void
+    public function configureComposerDependencies(DependencyManager $dependencyManager): void
     {
-        $dependencies->addClassDependency(AsTwigComponent::class, 'symfony/ux-twig-component');
+        // $this is a hack - we need the manager later in `interact()`
+        $this->dependencyManager = $dependencyManager;
+
+        $dependencyManager
+            ->addDependency(new RequiredClassDependency(
+                className: AsTwigComponent::class,
+                composerPackage: 'symfony/ux-twig-component',
+                preInstallMessage: 'This command requires the Symfony UX Twig Component Package.'
+            ))
+            ->addDependency(new OptionalClassDependency(
+                className: AsLiveComponent::class,
+                composerPackage: 'symfony/ux-live-component',
+                preInstallMessage: 'The Symfony UX Live Component is needed to make this a live component.'
+            ))
+        ;
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        if (!$input->getOption('live')) {
+            $input->setOption('live', $io->confirm('Make this a live component?'));
+        }
+
+        if (!$input->getOption('live')) {
+            return;
+        }
+
+        // @TODO - with the dependencyManager in `Command` -> we can't use it outside of configure dependencies.....
+        $this->dependencyManager->installOptionalDependencies();
```

Comment:
Not sure how to go about using the manager in other maker methods (aside from configureDepends...) Gonna play around with this a bit more. but other ideas are welcome.. Thought about some sort of setter in `AbstractMaker` or a constructor param in the individual makers that needs this...